tldw_Server_API/app/core/Explainer/chatbook_adapter.py

```
@@ -0,0 +1,549 @@
+    explainer_payload = _unwrap_explainer_payload(payload)
+    structured = explainer_payload.get("structured")
+    if not isinstance(structured, dict):
+        raise ValueError("Explainer payload missing structured content")
+
+    session_payload = structured.get("session")
+    if not isinstance(session_payload, dict):
+        raise ValueError("Explainer payload missing session")
+
+    nodes_payload = structured.get("nodes")
+    if not isinstance(nodes_payload, list) or not nodes_payload:
+        raise ValueError("Explainer payload missing nodes")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)